### PR TITLE
Tuple docs.

### PR DESCRIPTION
**Pull request by newkek**

### CODE REVIEW

File: manual/tuples/README.md [hunk 1]
```diff
@@ -1,5 +1,77 @@
...
+  ck1 text,
+  ck2 text,
+  v tuple<int, text, float>,
+  PRIMARY KEY (pk, ck1, ck2)
+);
+```
+
+### Fetching Tuples from Rows results
+
+The DataStax Java driver exposes a special `TupleValue` class to handle such columns. `TupleValue` extends `GettableByIndexData` class, which allows to call specific `get...(int)` methods on a returned `TupleValue`:
```
Comment (tolbertam) [inline manual/tuples/README.md]: Would be good to link to `TupleValue` & `GettableByIndexData` javadoc on the first reference to them.

**Review by tolbertam — APPROVED** [review]
Looks good.  Had a small suggestion for javadoc links.

File: manual/tuples/README.md [hunk 2]
```diff
@@ -1,5 +1,77 @@
-## Tuples
+## Using Tuples with the Java driver
 
-*Coming soon... In the meantime, see the javadoc for [TupleType].*
+Cassandra allows to use `tuple` data types [in tables](https://docs.datastax.com/en/cql/3.1/cql/cql_reference/tupleType.html):
```
Comment (adutra) [inline manual/tuples/README.md]: They are also allowed in user types.

File: manual/tuples/README.md [hunk 3]
```diff
@@ -1,5 +1,77 @@
...
+int firstValueInTuple = tupleValue.getInt(0);
+
+String secondValueInTuple = tupleValue.getString(1);
+
+Float thirdValueInTuple = tupleValue.getFloat(2);
+```
+
+### Using tuples as statement parameters
+
+A prepared statement may be containing a Tuple as a query parameter. In such case, users will need to create or gather a `TupleType` first, in order to be able to create a `TupleValue` to bind:
```
Comment (adutra) [inline manual/tuples/README.md]: Nit: "may contain" ... "In such cases,..."

File: manual/tuples/README.md [hunk 4]
```diff
@@ -1,5 +1,77 @@
...
+bs.setString("ck1", "1");
+bs.setString("ck2", "1");
+bs.setTupleValue("v", tupleType.newValue(1, "hello", 2.3f));
+
+session.execute(bs);
+```
+
+#### More use cases
+
+Users can also define single-usage tuples in _SELECT_ queries with the `IN` keyword, usually for tables with composite clustering keys, in that case a tuple will be usable the same way it was for prepared statements parameters:
```
Comment (adutra) [inline manual/tuples/README.md]: This is called a "multi-column IN restriction" – maybe it's worth mentioning this term somewhere.

**Review by adutra — CHANGES_REQUESTED** [review]

**Review by adutra — APPROVED** [review]

File: manual/tuples/README.md [hunk 5]
```diff
@@ -1,5 +1,87 @@
...
+  ck2 text,
+  v tuple<int, text, float>,
+  PRIMARY KEY (pk, ck1, ck2)
+);
+```
+
+### Fetching Tuples from Rows results
+
+The DataStax Java driver exposes a special [TupleValue] class to handle such columns. 
+[TupleValue] extends [GettableByIndexData] class, which allows to call specific `get...(int)`
```
Comment (olim7t) [inline manual/tuples/README.md]: nit: GettableByIndexData is an implementation detail, not sure if it's relevant to mention it here. All the user needs to know is that there are getter methods.

File: manual/tuples/README.md [hunk 6]
```diff
@@ -1,5 +1,87 @@
...
+```java
+PreparedStatement ps = session.prepare("INSERT INTO ks.collect_things (pk, ck1, ck2, v) VALUES (:pk, :ck1, :ck2, :v)");
+
+TupleType tupleType = cluster.getMetadata().newTupleType(DataType.cint(), DataType.text(), DataType.cfloat());
+
+BoundStatement bs = ps.bind();
+bs.setInt("pk", 1);
+bs.setString("ck1", "1");
+bs.setString("ck2", "1");
+bs.setTupleValue("v", tupleType.newValue(1, "hello", 2.3f));
```
Comment (olim7t) [inline manual/tuples/README.md:48]: Would be useful to mention that `newValue(Object...)` follows the same rules as `new SimpleStatement(String, Object...)`, i.e. type inference with some ambiguities (like numeric literals always interpreted as `int`).
Another way to create a tuple value -- and the workaround to solve those ambiguities -- is the parameterless `newValue()` followed by setter calls.

**Review by olim7t — CHANGES_REQUESTED** [review]

**Review by olim7t — APPROVED** [review]

**Comment by olim7t** [comment]
lgtm, but maybe include a code example of newValue() + setters as well?
(even though it's obvious)

**Pull request merged**